executor/src/witgen/jit/witgen_inference.rs

```diff
@@ -321,14 +324,15 @@ impl<'a, T: FieldElement, FixedEval: FixedEvaluator<T>> WitgenInference<'a, T, F
         (lhs_evaluated - rhs_evaluated).solve()
     }
 
-    fn process_call(
+    fn process_call_inner(
         &mut self,
         can_process_call: impl CanProcessCall<T>,
         lookup_id: u64,
         selector: &Expression<T>,
-        arguments: &'a [Expression<T>],
+        argument_count: usize,
         row_offset: i32,
     ) -> ProcessResult<T, Variable> {
+        self.process_assignments().unwrap();
```

Comment:
```suggestion
        // Make sure any known arguments are stored in the correct machine
        // call variable.
        self.process_assignments().unwrap();
```

Reply:
Ah this is actually a hack because it didn't work. I think it's just another manifestation of the bug I want to fix. I will delete this line once witgen_inference has no authority over the assignments any more.